### DISCUSSION TITLE
Run an executable with arguments from an extension command

### DISCUSSION THREAD

**Discussion by umar-shabbir** [microsoft/vscode-discussions#1000] [Extension Development QnA] [answered]
What is the best way to execute a shell command from an extension? The command will invoke an exe present on local file system

I have an extension which registers a command. The command is triggered by explorer context menu click. This command should run an executable present on local file system and also pass some arguments to that exe file. So that means it should run a shell command like:
program.exe --arg1 val1 --arg2 val2

The progress of above command on exe file needs to be tracked and its console output should be displayed on the terminal. Also there should be an option to cancel this running command.

How to implement it in VS code extension? I used the 
`let t = window.createTerminal();` and `t.sendText()` to create a terminal and execute command in it. The command runs in the terminal, output is displayed but there is no way to cancel the operation or track its progress and display a loading icon or progress bar somewhere. For cancellation, I want to add a button to cancel the running command in terminal.

**Comment by r3m0t** [comment, marked as answer]
`vscode.tasks.executeTask`
